File: test_translation.js
// 测试翻译服务是否正常工作
const https = require('https');
const http = require('http');

// 测试mymemory翻译服务
function testMymemory(text, src, target) {
    return new Promise((resolve, reject) => {
        const url = `https://api.mymemory.translated.net/get?q=${encodeURIComponent(text)}&langpair=${encodeURIComponent(src)}|${encodeURIComponent(target)}`;
        
        console.log('测试mymemory翻译服务:', url);
        
        https.get(url, (res) => {
            let data = '';
            
            res.on('data', (chunk) => {
                data += chunk;
            });
            
            res.on('end', () => {
                try {
                    const result = JSON.parse(data);
                    console.log('mymemory翻译结果:', result);
                    if (result.responseData && result.responseData.translatedText) {
                        resolve(result.responseData.translatedText);
                    } else {
                        reject(new Error('mymemory返回空结果'));
                    }
                } catch (e) {
                    reject(new Error(`解析mymemory响应失败: ${e.message}`));
                }
            });
        }).on('error', (e) => {
            reject(new Error(`mymemory请求失败: ${e.message}`));
        });
    });
}

// 测试libretranslate翻译服务
function testLibretranslate(text, src, target) {
    return new Promise((resolve, reject) => {
        const postData = JSON.stringify({
            q: text,
            source: src,
            target: target,
            format: 'text'
        });
        
        const options = {
            hostname: 'libretranslate.com',
            port: 443,
            path: '/translate',
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
                'Content-Length': Buffer.byteLength(postData)
            }
        };
        
        console.log('测试libretranslate翻译服务:', options.hostname + options.path);
        
        const req = https.request(options, (res) => {
            let data = '';
            
            res.on('data', (chunk) => {
                data += chunk;
            });
            
            res.on('end', () => {
                try {
                    const result = JSON.parse(data);
                    console.log('libretranslate翻译结果:', result);
                    if (result.translatedText) {
                        resolve(result.translatedText);
                    } else {
                        reject(new Error('libretranslate返回空结果'));
                    }
                } catch (e) {
                    reject(new Error(`解析libretranslate响应失败: ${e.message}`));
                }
            });
        });
        
        req.on('error', (e) => {
            reject(new Error(`libretranslate请求失败: ${e.message}`));
        });
        
        req.write(postData);
        req.end();
    });
}

// 运行测试
async function runTests() {
    console.log('开始测试翻译服务...');
    
    const testText = 'Hello, world!';
    const srcLang = 'en';
    const targetLang = 'ru';
    
    try {
        console.log('\n1. 测试mymemory翻译服务:');
        const mymemoryResult = await testMymemory(testText, srcLang, targetLang);
        console.log('✓ mymemory翻译成功:', testText, '->', mymemoryResult);
    } catch (e) {
        console.log('✗ mymemory翻译失败:', e.message);
    }
    
    try {
        console.log('\n2. 测试libretranslate翻译服务:');
        const libreResult = await testLibretranslate(testText, srcLang, targetLang);
        console.log('✓ libretranslate翻译成功:', testText, '->', libreResult);
    } catch (e) {
        console.log('✗ libretranslate翻译失败:', e.message);
    }
    
    console.log('\n测试完成!');
}

runTests();

File: test_translation.py
import requests
import json
import time

def test_translation():
    print('测试俄语翻译功能')
    
    try:
        # 测试文本
        text = "You didn't hear that?"
        
        print(f"准备翻译文本: '{text}'")
        print(f"目标语言: ru")
        
        # 构建请求数据
        request_data = {
            'items': [{ 'text': text }],
            'target': 'ru'
        }
        
        print(f"请求数据: {json.dumps(request_data)}")
        
        # 尝试调用翻译服务
        print("发送请求到 http://localhost:8000/translate...")
        response = requests.post(
            'http://localhost:8000/translate',
            headers={
                'Content-Type': 'application/json'
            },
            data=json.dumps(request_data),
            timeout=30  # 设置30秒超时
        )
        
        print(f"收到响应，状态码: {response.status_code}")
        print(f"响应头: {dict(response.headers)}")
        
        if not response.ok:
            print(f"响应内容: {response.text}")
            raise Exception(f'HTTP错误! 状态: {response.status_code}, 内容: {response.text}')
        
        result = response.json()
        print(f"响应内容: {json.dumps(result, indent=2, ensure_ascii=False)}")
        
        if result and result['ok'] and result['items'] and len(result['items']) > 0:
            print(f'翻译成功: "{text}" -> "{result["items"][0]["text"]}"')
        else:
            print(f'翻译失败: {json.dumps(result, ensure_ascii=False)}')
        
    except requests.exceptions.RequestException as e:
        print(f'网络请求失败: {type(e).__name__}: {e}')
        import traceback
        traceback.print_exc()
    except json.JSONDecodeError as e:
        print(f'JSON解析失败: {e}')
        import traceback
        traceback.print_exc()
    except Exception as error:
        print(f'测试失败: {type(error).__name__}: {error}')
        import traceback
        traceback.print_exc()

if __name__ == '__main__':
    print('开始测试...')
    test_translation()
    print('测试结束.')